features/mbedtls/targets/TARGET_NUVOTON/TARGET_M480/aes/aes_alt.c

```diff
@@ -21,298 +21,220 @@
+     * (1) Word-aligned buffer base address
+     * (2) 16-byte aligned buffer size
+     * (3) Located in 0x20000000-0x2FFFFFFF region
+     */
+    if ((! crypto_dma_buff_compat(au8OutputData, MAX_DMA_CHAIN_SIZE, 16)) ||
+        (! crypto_dma_buff_compat(au8InputData, MAX_DMA_CHAIN_SIZE, 16))) {
+        error("Buffer for AES alter. DMA requires to be word-aligned and located in 0x20000000-0x2FFFFFFF region.");
+    }
+
+    /* TODO: Change busy-wait to other means to release CPU */
+    /* Acquire ownership of AES H/W */
+    while (! crypto_aes_acquire());
+    
+    /* Init crypto module */
+    crypto_init();
+    /* Enable AES interrupt */
+    AES_ENABLE_INT();
+    
+    /* We support multiple contexts with context save & restore and so needs just one 
+     * H/W channel. Always use H/W channel #0. */
+
+    /* AES_IN_OUT_SWAP: Let H/W know both input/output data are arranged in little-endian */
+    AES_Open(0, ctx->encDec, ctx->opMode, ctx->keySize, AES_IN_OUT_SWAP);
+    AES_SetInitVect(0, ctx->iv);
+    AES_SetKey(0, ctx->keys, ctx->keySize);
+    /* AES DMA buffer requirements same as above */
+    if (! crypto_dma_buff_compat(input, dataSize, 16)) {
+        memcpy(au8InputData, input, dataSize);
+        pIn = au8InputData;
+    } else {
+        pIn = input;
+    }
+    /* AES DMA buffer requirements same as above */
+    if (! crypto_dma_buff_compat(output, dataSize, 16)) {
+        pOut = au8OutputData;
+    } else {
+        pOut = output;
     }
-		if( (((uint32_t)output) & 0x03) || (dataSize%4))   // HW CFB output byte count must be multiple of word
-		{		
-			pOut = au8OutputData;
-		} else {
-		  pOut = output;
-    }			
 
-    AES_SetDMATransfer(ctx->channel, (uint32_t)pIn, (uint32_t)pOut, dataSize);		
+    AES_SetDMATransfer(0, (uint32_t)pIn, (uint32_t)pOut, dataSize);
 
     g_AES_done = 0;
-    AES_Start(ctx->channel, CRYPTO_DMA_ONE_SHOT);
+    AES_Start(0, CRYPTO_DMA_ONE_SHOT);
     while (!g_AES_done);
-
-    if( pOut != output ) memcpy(output, au8OutputData, dataSize);
-		dumpHex(output,16);
-
+    
+    if( pOut != output ) {
+        memcpy(output, au8OutputData, dataSize);
+    }
+    
+    /* Save IV for next block */
+    ctx->iv[0] = CRPT->AES_FDBCK[0];
+    ctx->iv[1] = CRPT->AES_FDBCK[1];
+    ctx->iv[2] = CRPT->AES_FDBCK[2];
+    ctx->iv[3] = CRPT->AES_FDBCK[3];
+    
+    /* Disable AES interrupt */
+    AES_DISABLE_INT();
+    /* Uninit crypto module */
+    crypto_uninit();
+    
+    /* Release ownership of AES H/W */
+    crypto_aes_release();
 }
 
 /*
  * AES-ECB block encryption
  */
-#if defined(MBEDTLS_AES_ENCRYPT_ALT)
 void mbedtls_aes_encrypt( mbedtls_aes_context *ctx,
                           const unsigned char input[16],
                           unsigned char output[16] )
 {
-
-	  mbedtls_trace("=== %s \r\n", __FUNCTION__);
-	
-	  ctx->encDec = 1;
-	  __nvt_aes_crypt(ctx, input, output, 16);
-  
+    ctx->encDec = 1;
+    __nvt_aes_crypt(ctx, input, output, 16);
```

Comment:
@Patater Yes, it would be better to replace 16 with `MBEDTLS_AES_BLOCK_SIZE` (via `#define`) for readibility. But I find original mbed TLS has coded with 16 (`mbed-os/features/mbedtls/inc/mbedtls/aes.h` and `mbed-os/features/mbedtls/src/aes.c`). To make code consistency, I follow the style and don't replace 16 and the like with macro.